src/Symfony/Component/String/UnicodeString.php

```diff
@@ -332,7 +336,15 @@ public function startsWith($prefix): bool
         $form = null === $this->ignoreCase ? \Normalizer::NFD : \Normalizer::NFC;
         normalizer_is_normalized($prefix, $form) ?: $prefix = normalizer_normalize($prefix, $form);
 
-        return '' !== $prefix && false !== $prefix && 0 === ($this->ignoreCase ? grapheme_stripos($this->string, $prefix) : grapheme_strpos($this->string, $prefix));
+        if ('' === $prefix || false === $prefix) {
+            return false;
+        }
+
+        if ($this->ignoreCase) {
+            return 0 === grapheme_stripos($this->string, $prefix);
```

Comment:
Maybe this could be improved by using `grapheme_extract` and then `grapheme_stripos` against the extracted part.
@nicolas-grekas What do you think, is it worth it?
Are there graphemes with different byte length in lowercased and uppercased variant?
(the german `ß` in uppercase is `SS`, but this [is not matched by `*_stripos`](https://3v4l.org/Kf8rZ), so it does not matter here.)

Reply:
a bench would tell... :)

Reply:
looks good: https://3v4l.org/QeEdo